pkg/server/webhook.go

```diff
@@ -420,6 +430,10 @@ func createPatch(pod *corev1.Pod, inj *config.InjectionConfig, annotations map[s
 	patch = append(patch, addContainers(pod.Spec.InitContainers, mutatedInjectedInitContainers, "/spec/initContainers")...)
 	patch = append(patch, addHostAliases(pod.Spec.HostAliases, inj.HostAliases, "/spec/hostAliases")...)
 	patch = append(patch, addVolumes(pod.Spec.Volumes, inj.Volumes, "/spec/volumes")...)
+	if inj.ServiceAccountName != "" && pod.Spec.ServiceAccountName == "" {
+		// only override the serviceaccount name if not set in the pod spec
```

Comment:
is this documented some place beside this code spot?

Reply:
Ill update the docs as well. Good catch